cibuildwheel/util.py

```diff
@@ -18,13 +18,7 @@
 from .environment import ParsedEnvironment
 from .typing import PathOrStr, PlatformName
 
-if sys.version_info < (3, 9):
-    from importlib_resources import files
-else:
-    from importlib.resources import files
-
-
-resources_dir = files('cibuildwheel') / 'resources'
+resources_dir = Path(__file__).parent / 'resources'
```

Comment:
Why this direction? This was the "correct" language to access resources using the stdlib (and backport in <3.9). It follows the standard import mechanisms instead of relying on manual path manipulations. We don't load Python files by path, why load data that way?

And I believe the rewritten version is misleading as it stands, without a `.resolve()` it doesn't work from a main file before 3.{8,9,10}, don't remember exactly which one they started always resolving the `__file__` in to protect against changed directories.

Reply:
I couldn't get mypy to accept the importlib_resources version, for some reason. So I put it back to the way it used to be. Btw, I don't think that we can get away from the fact that resources are files, since we get their paths to pass them to subprocesses, or we run them from the filesystem like `get-pip`. I'd rather use simpler concepts like Paths if there's no reason not to.

Reply:
> rather use simpler concepts

I still think it's better to use the correct concept, rather than propagating the idea that it's okay to manipulate paths inside a package. I'm find with path manipulations in `bin`, but I don't like them in `cibuildwheel`.

> since we get their paths to pass them to subprocesses, or we run them from the filesystem like get-pip.

importlib.resouces has a method for doing this, though it's one extra step. If you add a context manager, it will promise a real path (to somewhere, either the actual location or a temporary location for some loaders) for the duration. I didn't go through and update the usages of pure pathnames for this - but in theory, if we did, we could then make a cibuldwheel.pyz zip file that could be run from anywhere without an internet connection, for example.

Because we aren't careful with this, it's mostly a conceptual improvement for the moment, so if you wanted to revert it for now, but ideally we could do it correctly some day. cibuildwheel does have the distinction of simply needing to run on CI rather than absolutely everywhere.

> I couldn't get mypy to accept the importlib_resources version, for some reason

The MyPy improvements in #536 touch on this partially. It's generally a little odd for backports, since MyPy assumes the target Python version but usually runs on the latest Python version.

Reply:
> rather than propagating the idea that it's okay to manipulate paths inside a package

But the resources are files, right? They need to be files, otherwise we'd have to copy them out to a temp dir before we could use them?

Reply:
Opening a resource works just like normal files, even if, for example, are coming from a zip file because Python supports site-packages being a zip file. If you need an actual file name, you do:

```
from importlib.resources import as_file
with as_file(package_dir / "my_file.txt") as file:
    third_party_api_requiring_file_system_path(file)
```
For the duration of that context manager, the resource is actually a real file (either the normal one from the file system, or one extracted to a tmp dir if the original loader (like the zip loader) doesn't provide file names.

https://importlib-resources.readthedocs.io/en/latest/using.html#file-system-or-zip-file

Reply:
So until we add the above, some parts of cibuildwheel do not support running from arbitrary loaders, since we do require file names in a few places. So there's only a conceptual benefit (not showing bad examples of path manipulation in a library) to using importlib.resouces until we do that (at which time, there would be a real benefit).